SU-SWS/decanter
# Add a variant for Card where the whole card is a link and add option to turn headline into a link

# READY FOR REVIEW

# Summary
- Add a variant for the Card component where the whole card is a link (works for both default and horizontal versions)
- There's logic in the twig template that does not render button link nor CTA link in this variant.
- Fix accessibility issue found by Level Access where h4 (superheadline) was above the h3 (headline) by changing h4 to a span, since it isn't really semantically a headline anyway.
- Add option to turn headline into a link

# Needed By (Date)
- By end of week

# Urgency
- ?

# Steps to Test

1. Pull this branch, do `npm run build` and `npm run styleguide`.
2. Look at the `.su-card--link` variant and see if it looks and behaves ok including at mobile breakpoints. Check that the markup is semantic.
3. Check that the headline is now a link in all the variants except `.su-card--link` (we passed in the variable `card_headline_link = true` in the .json)
4. Add `"modifier_class": "su-card--horizontal su-card--link", ` to card.json, save the file and do `npm run styleguide`.
5. Look at the card component again - they should all render as horizontal cards where whole card is  link now. Check that it looks ok also. 

# Affected Projects or Products
- Decanter

# Associated Issues and/or People
- https://github.com/SU-SWS/decanter/issues/430
- https://github.com/SU-SWS/decanter/issues/407
- #453
